packages/java/endpoint/src/main/java/dev/hilla/internal/hotswap/HotSwapConfiguration.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2000-2023 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package dev.hilla.internal.hotswap;
+
+import dev.hilla.EndpointController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class HotSwapConfiguration {
+
+    /**
+     * The interval for polling the changes in seconds.
+     */
+    @Value("${hilla.endpoint.hot-reload.pollInterval:5}")
```

Comment:
You should aim to declare properties in a way so that tooling can pick it up and offer auto complete, like in https://github.com/vaadin/flow/blob/main/vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinConfigurationProperties.java

Reply:
Done